src/types.ts

```diff
@@ -23,7 +23,7 @@ export interface KnownBaseTheme {
     [key: string]: string;
   };
   spacing: {
-    [key: string]: number;
+    [key: string]: number | 'auto';
```

Comment:
I think it might be better if used the same type as React Native, may be not `undefined`
```
margin?: number | string | undefined;
    marginBottom?: number | string | undefined;
    marginEnd?: number | string | undefined;
    marginHorizontal?: number | string | undefined;
    marginLeft?: number | string | undefined;
    marginRight?: number | string | undefined;
    marginStart?: number | string | undefined;
    marginTop?: number | string | undefined;
    marginVertical?: number | string | undefined;
```

```suggestion
    [key: string]: number | string;
```